### PR TITLE
cli generators tests

### PR DESCRIPTION
**Pull request by KavishShah09**

### CODE REVIEW

File: space-cli/cmd/utils/input/input.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// AskOne calls survey.AskOne
+func (i *Input) AskOne(p survey.Prompt, respone interface{}, opts ...survey.AskOpt) error {
+	if err := survey.AskOne(p, respone, opts...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Survey package for survey.Askone
+var Survey inputInterface
```
Comment (sharadregoti) [inline space-cli/cmd/utils/input/input.go:21]: why do we need this ?
take a look at this pr https://github.com/spaceuptech/space-cloud/pull/995/files

Comment (KavishShah09) [reply space-cli/cmd/utils/input/input.go:21]: Check the test example in the below commit

File: space-cli/cmd/modules/letsencrypt/generate_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				API:  "/v1/config/projects/{project}/letsencrypt/config/{id}",
+				Type: "letsencrypt",
+				Meta: map[string]string{
+					"project": "",
+					"id":      "letsencrypt-config",
+				},
+				Spec: map[string]interface{}{
+					"domains": []string{"domain1", "domain2"},
+				},
+			},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/letsencrypt/generate_test.go:79]: And one more functionality in generateletsencrypt. Ask for email id after getting the domain name. Add the above in test also.
Check letsencrypt config struct in gateway for more info

File: space-cli/cmd/modules/letsencrypt/generate_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spaceuptech/space-cli/cmd/model"
+	"github.com/spaceuptech/space-cli/cmd/utils"
+	"github.com/spaceuptech/space-cli/cmd/utils/input"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_generateLetsEncryptDomain(t *testing.T) {
+	someString := ""
```
Comment (sharadregoti) [inline space-cli/cmd/modules/letsencrypt/generate_test.go:16]: Give a useful name to this string. Do this refactoring everywhere in this pr/

File: space-cli/cmd/modules/project/generate_test.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spaceuptech/space-cli/cmd/model"
+	"github.com/spaceuptech/space-cli/cmd/utils"
+	"github.com/spaceuptech/space-cli/cmd/utils/input"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_generateProject(t *testing.T) {
+	someString := ""
+	contextTime := 0
```
Comment (sharadregoti) [inline space-cli/cmd/modules/project/generate_test.go:17]: Write a comment, answering why the variable is initialized with 0

File: space-cli/cmd/modules/project/generate_test.go [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+				API:  "/v1/config/projects/{project}",
+				Type: "project",
+				Meta: map[string]string{
+					"project": "project",
+				},
+				Spec: map[string]interface{}{
+					"id":                 "project",
+					"aesKey":             "key",
+					"name":               "projectName",
+					"secrets":            "",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/project/generate_test.go:147]: Add another test case in which secrets is specified

File: space-cli/cmd/modules/remote-services/generate_test.go [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spaceuptech/space-cli/cmd/model"
+	"github.com/spaceuptech/space-cli/cmd/utils"
+	"github.com/spaceuptech/space-cli/cmd/utils/input"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_generateService(t *testing.T) {
+	someString := ""
+	want := "y"
```
Comment (sharadregoti) [inline space-cli/cmd/modules/remote-services/generate_test.go:17]: Give a useful name & comment why it has been intilized with "y"

File: space-cli/cmd/modules/remote-services/generate_test.go [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+				Meta: map[string]string{
+					"id":      "",
+					"project": "",
+				},
+				Spec: map[string]interface{}{
+					"url":       "",
+					"endpoints": []interface{}{map[string]interface{}{"endpointName": map[string]interface{}{"method": "", "path": ""}}},
+				},
+			},
+		},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/remote-services/generate_test.go:262]: All the variables cannot be empty string

File: space-cli/cmd/modules/userman/generate_test.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+				Meta: map[string]string{
+					"project": "",
+					"id":      "",
+				},
+				Spec: map[string]interface{}{
+					"enabled": true,
+					"secret":  "",
+				},
+			},
+		},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/userman/generate_test.go:76]: Once again all these variables cannot be empty, do the changes everywhere in this pr

File: space-cli/cmd/utils/input/input.go [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+type inputInterface interface {
+	AskOne(p survey.Prompt, respone interface{}, opts ...survey.AskOpt) error
+}
+
+// Input struct for parameter
+type input struct{}
+
+// AskOne calls survey.AskOne
+func (i *input) AskOne(p survey.Prompt, respone interface{}, opts ...survey.AskOpt) error {
```
Comment (sharadregoti) [inline space-cli/cmd/utils/input/input.go:13]: I believe the 3rd parameter is useless. As we are not using it anywhere in the test case. Verify it & if we are not using remote it. Directly pass it to survey.AskOne

File: space-cli/cmd/utils/types.go [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+	"github.com/stretchr/testify/mock"
+)
+
+// MockInputInterface is the mock interface for survey package
+type MockInputInterface struct {
+	mock.Mock
+}
+
+// AskOne is the mock method for MockInputInterface
+func (m *MockInputInterface) AskOne(p survey.Prompt, response interface{}, opts ...survey.AskOpt) error {
```
Comment (sharadregoti) [inline space-cli/cmd/utils/types.go:16]: same here

File: space-cli/cmd/modules/services/generate_test.go [hunk 11]
```diff
@@ -0,0 +1,531 @@
...
+							Runtime:   model.Image,
+							Secrets:   []string{},
+							Env:       map[string]string{},
+						},
+					},
+					Affinity:  []model.Affinity{},
+					Whitelist: []model.Whitelist{{ProjectID: "projectID", Service: "*"}},
+					Upstreams: []model.Upstream{{ProjectID: "projectID", Service: "*"}},
+				},
+			},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/services/generate_test.go:503]: Fields cannot be empty

File: space-cli/cmd/modules/database/generate_test.go [hunk 12]
```diff
@@ -0,0 +1,616 @@
...
+			},
+			want: &model.SpecObject{
+				API:  "/v1/config/projects/{project}/database/{dbAlias}/collections/{col}/rules",
+				Type: "db-rules",
+				Meta: map[string]string{
+					"dbAlias": "",
+					"col":     "",
+					"project": "",
+				},
+				Spec: map[string]interface{}{
```
Comment (sharadregoti) [inline space-cli/cmd/modules/database/generate_test.go:170]: fields cannot be empty

**Pull request merged**